File: Common/Reports.fs
module Reports
open System
open DistributedTaskDataContracts.Messages.Reports.FalseCatcher
open DistributedTaskDataContracts.Messages.Reports.Scanner
open DistributedTaskDataContracts
open CommonDataStructures.Scanner
open KlSrl.Core.PluginsConfiguration

let lastTaskId = ref 0L

let nextTaskId() = 
    lastTaskId := !lastTaskId + 1L
    !lastTaskId

module Fc = 
    let private pluginGuid = PluginGuids.GetPluginGuid(PluginGuids.FalseCatcherPlugin)
    let success() = SuccessFalseCatcherReport(CompositeTaskId(nextTaskId()), PluginGuid = pluginGuid)
    let failed() = FailedFalseCatcherReport(CompositeTaskId(nextTaskId()), "Error message 1", PluginGuid = pluginGuid)

module ReScan = 
    let private pluginGuid = PluginGuids.GetPluginGuid(PluginGuids.ReScannerPlugin)
    let private scanResult = FileScanResult([| DetectInfo("comment 1", Nullable 1L) |], true, false, "Error reason 1")
    let success() = SuccessSingleFileScannerReport(CompositeTaskId(nextTaskId()), scanResult, PluginGuid = pluginGuid)
    let failed() = FailedScannerReport(CompositeTaskId(nextTaskId()), "Error message 1", PluginGuid = pluginGuid)

File: Publisher/EasyNetQPublisher2/Reports.cs
//using System;
//using System.Threading;
//using CommonDataStructures.Scanner;
//using DistributedTaskDataContracts;
//using DistributedTaskDataContracts.Messages.Reports;
//using DistributedTaskDataContracts.Messages.Reports.FalseCatcher;
//using DistributedTaskDataContracts.Messages.Reports.Scanner;
//using KlSrl.Core.PluginsConfiguration;
//
//namespace EasyNetQPublisher2
//{
//    public static class Reports
//    {
//        static long _lastTaskId;
//
//        static long NextTaskId()
//        {
//            return Interlocked.Increment(ref _lastTaskId);
//        }
//
//        public static class Fc
//        {
//            static readonly Guid PluginGuid = PluginGuids.GetPluginGuid(PluginGuids.FalseCatcherPlugin);
//
//            public static PluginReport Success()
//            {
//                return new SuccessFalseCatcherReport(new CompositeTaskId(NextTaskId())) {PluginGuid = PluginGuid};
//            }
//
//            public static PluginReport Failed()
//            {
//                return new FailedFalseCatcherReport(new CompositeTaskId(NextTaskId()), "Error message 1")
//                    {
//                        PluginGuid = PluginGuid
//                    };
//            }
//        }
//
//        public static class ReScan
//        {
//            static readonly Guid PluginGuid = PluginGuids.GetPluginGuid(PluginGuids.ReScannerPlugin);
//
//            public static PluginReport Success()
//            {
//                var scanResult = new FileScanResult(new[] {new DetectInfo("comment 1", 1)}, true, false,
//                                                    "Error reason 1");
//                return new SuccessSingleFileScannerReport(new CompositeTaskId(NextTaskId()), scanResult)
//                    {
//                        PluginGuid = PluginGuid
//                    };
//            }
//
//            public static PluginReport Failed()
//            {
//                return new FailedScannerReport(new CompositeTaskId(NextTaskId()), "Error message 1")
//                    {
//                        PluginGuid = PluginGuid
//                    };
//            }
//        }
//    }
//}
